app/models/commit.rb

```diff
@@ -88,7 +88,12 @@ def add_status(status_attributes)
     statuses.create!(status_attributes)
     reload # to get the statuses into the right order (since sorted :desc)
     new_status = significant_status
-    Hook.emit(:commit_status, stack, commit: self, stack: stack, status: new_status) if previous_status != new_status
+    Hook.emit(:commit_status, stack, commit: self, stack: stack, commit_status: new_status,
+                                     status: new_status.state) if previous_status != new_status
+    if simple_state(previous_status) != simple_state(new_status) && new_status.state != 'pending'
+      Hook.emit(:deployable_status, stack, commit: self, stack: stack, deployable_status: new_status,
+                                           status: new_status.state)
+    end
```

Comment:
Maybe be it would be lighter to the eye, in an early return style:

```ruby
return if new_status == previous_status
Hook.emit(:commit_status, stack, ...)

return if new_status.pending?
return if simple_state(previous_status) == simple_state(new_status)
Hook.emit(:deployable_status, stack, ...)
```

What do you think?